backend/src/DataAccessLayer/dalMain.cs

```diff
@@ -35,16 +35,53 @@ public CosmosDal(Secrets secrets, Config config)
+                CosmosLease = secrets.CosmosLease,
                 CosmosDatabase = secrets.CosmosDatabase,
                 CosmosKey = secrets.CosmosKey,
                 CosmosUrl = secrets.CosmosServer,
                 Retries = config.Retries,
                 Timeout = config.Timeout,
             };
 
-            // create the CosmosDB client and container
-            cosmosDetails.Client = OpenAndTestCosmosClient(secrets.CosmosServer, secrets.CosmosKey, secrets.CosmosDatabase, secrets.CosmosCollection).GetAwaiter().GetResult();
-            cosmosDetails.Container = cosmosDetails.Client.GetContainer(secrets.CosmosDatabase, secrets.CosmosCollection);
+            // create the CosmosDB client and source container
+            cosmosDetails.Client = new CosmosClient(secrets.CosmosServer, secrets.CosmosKey, cosmosDetails.CosmosClientOptions);
+            cosmosDetails.SourceContainer = cosmosDetails.Client.GetContainer(secrets.CosmosDatabase, secrets.CosmosCollection);
+            cosmosDetails.LeaseContainer = cosmosDetails.Client.GetContainer(secrets.CosmosDatabase, secrets.CosmosLease);
+            cosmosDetails.ChangeFeedProcessor = StartChangeFeedProcessorAsync(cosmosDetails).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Start the Change Feed Processor to listen for changes and process them with the HandleChangesAsync implementation
+        /// </summary>
+        /// <param name="cosmosDetails">Cosmos details</param>
+        /// <returns>ChangeFeedProcessor</returns>
+        private static async Task<ChangeFeedProcessor> StartChangeFeedProcessorAsync(CosmosConfig cosmosDetails)
+        {
+            ChangeFeedProcessor changeFeedProcessor = cosmosDetails.SourceContainer
+                .GetChangeFeedProcessorBuilder<ClientStatus>(processorName: "clientStatusProcessor", onChangesDelegate: HandleChangesAsync)
+                .WithInstanceName("clientStatusProcessorInstance")
+                .WithLeaseContainer(cosmosDetails.LeaseContainer)
+                .Build();
+
+            Console.WriteLine("Starting Change Feed Processor...");
+            await changeFeedProcessor.StartAsync();
+            Console.WriteLine("Change Feed Processor started.");
+            return changeFeedProcessor;
+        }
+
+        /// <summary>
+        /// The delegate receives batches of changes as they are generated in the change feed and can process them
+        /// </summary>
+        private static async Task HandleChangesAsync(IReadOnlyCollection<ClientStatus> changes, CancellationToken cancellationToken)
+        {
+            Console.WriteLine("Started handling changes...");
+            foreach (ClientStatus item in changes)
+            {
+                Console.WriteLine($"Detected operation for item with id {item.Id}, last updated at {item.LastUpdated}.");
+                await Task.Delay(10);
+            }
+
+            Console.WriteLine("Finished handling changes.");
         }
```

Comment:
I'm wondering if we can make this a little more generic from a changefeed setup standpoint and use it for all setups.  Let's find time to look at it together and discuss.  Also, we should make sure we're accounting for the need to have an iterator that has a filter query.

Reply:
From our discussion, leave as-is and we'll change it in the Loderunner data access.